src/ralph/backends/data/clickhouse.py

```diff
@@ -382,6 +382,22 @@ def write(  # pylint: disable=too-many-arguments
 
         return count
 
+    def close(self) -> None:
+        """Close the ClickHouse backend client.
+
+        Raise:
+            BackendException: If a failure during the close operation occurs.
+        """
+        if not self._client:
```

Comment:
Perhaps we can add a warning to say that no client has been created ?

(applicable to all `close` methods)